From e0b7961c95f58079c789518a902727b7b68e0cb3 Mon Sep 17 00:00:00 2001
From: Sam Lantinga <slouken@libsdl.org>
Date: Mon, 23 Mar 2020 11:42:44 -0700
Subject: [PATCH] Fixed building back to Mac OSX using the 10.7 SDK

---
 src/video/cocoa/SDL_cocoawindow.m | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/video/cocoa/SDL_cocoawindow.m b/src/video/cocoa/SDL_cocoawindow.m
index b5b19528c..297fef09e 100644
--- a/src/video/cocoa/SDL_cocoawindow.m
+++ b/src/video/cocoa/SDL_cocoawindow.m
@@ -1813,6 +1813,13 @@ Cocoa_SetWindowFullscreen(_THIS, SDL_Window * window, SDL_VideoDisplay * display
            This is no longer needed as of Mac OS X 10.15, according to bug 4822.
          */
         NSProcessInfo *processInfo = [NSProcessInfo processInfo];
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 101000 /* NSOperatingSystemVersion added in the 10.10 SDK */
+        typedef struct {
+            NSInteger majorVersion;
+            NSInteger minorVersion;
+            NSInteger patchVersion;
+        } NSOperatingSystemVersion;
+#endif
         NSOperatingSystemVersion version = { 10, 15, 0 };
         if (![processInfo respondsToSelector:@selector(isOperatingSystemAtLeastVersion:)] ||
             ![processInfo isOperatingSystemAtLeastVersion:version]) {
-- 
2.27.0

